src/ring_theory/adjoin_root.lean

```diff
@@ -641,6 +642,68 @@ by rw [quot_adjoin_root_equiv_quot_polynomial_quot, ring_equiv.symm_trans_apply,
     quot_map_C_map_span_mk_equiv_quot_map_C_quot_map_span_mk_symm_quot_quot_mk,
     quot_map_of_equiv_quot_map_C_map_span_mk_symm_mk]
 
+/-- Promote `quot_adjoin_root_equiv_quot_polynomial_quot` to an alg_equiv  -/
+noncomputable def quot_equiv_quot_map (f : R[X]) (I : ideal R) :
+  ((adjoin_root f) ⧸ (ideal.map (of f) I)) ≃ₐ[R]
+     ((R ⧸ I) [X]) ⧸ (ideal.span ({polynomial.map I^.quotient.mk f} : set ((R ⧸ I) [X]))) :=
+alg_equiv.of_ring_equiv (quot_adjoin_root_equiv_quot_polynomial_quot I f)
+begin
+  intros x,
+  have : algebra_map R (adjoin_root f ⧸ (ideal.map (of f) I)) x =
+    ideal.quotient.mk (ideal.map (adjoin_root.of f) I) ((mk f) (C x)) := rfl,
+  rw [this, quot_adjoin_root_equiv_quot_polynomial_quot_mk_of, map_C],
+  refl,
+end
+
+@[simp]
+lemma quot_equiv_quot_map_apply (f g : polynomial R) (I : ideal R)  :
```

Comment:
I am not completely sure, but can `simps` produce these two lemmas?

Reply:
So `simps` does produce some lemmas, but not the rights ones. Is it still worth adding `simps`?

Reply:
It's OK to keep the explicit ones, thanks.